### PR TITLE
Adjust CI workflows for Spack forks

### PR DESCRIPTION
**Pull request by wyphan**
Changes the following CI workflows such that they only run for the main repo `spack/spack`:
- Containers `workflows/build-containers.yml`
- MacOS builds nightly `workflows/macos_python.yml`
- Bootstrap `workflows/bootstrap.yml`

### CODE REVIEW

File: .github/workflows/build-containers.yml [hunk 1]
```diff
@@ -43,6 +43,7 @@ jobs:
                      [ubuntu-focal, 'linux/amd64,linux/arm64,linux/ppc64le', 'ubuntu:20.04'],
                      [ubuntu-jammy, 'linux/amd64,linux/arm64,linux/ppc64le', 'ubuntu:22.04']]
     name: Build ${{ matrix.dockerfile[0] }}
+    if: github.action_repository == 'spack/spack' || github.event != 'schedule'
```
Comment (alalazo) [inline .github/workflows/build-containers.yml]: `github.action_repository` seems the wrong attribute. I'd also prefer to block this if not targeted at the main repo:
```suggestion
    if: github.repository == 'spack/spack'
```

File: .github/workflows/macos_python.yml [hunk 2]
```diff
@@ -22,6 +22,7 @@ on:
 jobs:
   install_gcc:
     name: gcc with clang
+    if: github.action_repository == 'spack/spack' || github.event != 'schedule'
```
Comment (alalazo) [inline .github/workflows/macos_python.yml]:
```suggestion
    if: github.repository == 'spack/spack'
```

File: .github/workflows/macos_python.yml [hunk 3]
```diff
@@ -36,6 +37,7 @@ jobs:
 
   install_jupyter_clang:
     name: jupyter
+    if: github.action_repository == 'spack/spack' || github.event != 'schedule'
```
Comment (alalazo) [inline .github/workflows/macos_python.yml]:
```suggestion
    if: github.repository == 'spack/spack'
```

File: .github/workflows/macos_python.yml [hunk 4]
```diff
@@ -50,6 +52,7 @@ jobs:
 
   install_scipy_clang:
     name: scipy, mpl, pd
+    if: github.action_repository == 'spack/spack' || github.event != 'schedule'
```
Comment (alalazo) [inline .github/workflows/macos_python.yml]:
```suggestion
    if: github.repository == 'spack/spack'
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Similar modifications to `bootstrap.yaml` are missing.

**Review by alalazo — APPROVED** [review]

**Comment by adamjstewart** [comment]
Can we also do this for dependabot? https://github.com/adamjstewart/spack/pulls?q=is%3Apr+is%3Aclosed

**Pull request merged**